test/NITTA/LuaFrontend/Tests.hs

```diff
@@ -22,60 +23,273 @@ module NITTA.LuaFrontend.Tests (
+        expectedValues = ("sum", [Lua.Name "a", Lua.Name "b"], [Lua.Number Lua.IntNum "1", Lua.Number Lua.IntNum "2"])
+     in (actualName, actualArg, actualArgValue) @?= expectedValues
+
+case_find_startup_function_no_args =
+    let src =
+            [__i|
+                function sum()
+                    t = receive()
+                    t = t + 1
+                    send(t)
+                    sum()
+                end
+                sum()
+            |]
+        (actualName, Lua.FunCall (Lua.NormalFunCall _ (Lua.Args actualArgValue)), Lua.FunAssign _ (Lua.FunBody actualArg _ _)) = findStartupFunction (getLuaBlockFromSources src)
+        expectedValues = ("sum", [], [])
+     in (actualName, actualArg, actualArgValue) @?= expectedValues
+
+-- |local a = 2
+case_process_local_assignment_statement =
+    let assignment = Lua.LocalAssign [Lua.Name "a"] (Just [Lua.Number Lua.IntNum "2"])
+        expected = HM.fromList [("a", LuaValueInstance{luaValueInstanceName = "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})]
+        (_, LuaAlgBuilder{algLatestLuaValueInstance}) = runState (processStatement "_" assignment) defaultAlgBuilder
+     in algLatestLuaValueInstance @?= expected
+
+-- |a = 2
+case_process_assignment_statement =
+    let assignment = Lua.Assign [Lua.VarName (Lua.Name "a")] [Lua.Number Lua.IntNum "2"]
+        expected = HM.fromList [("a", LuaValueInstance{luaValueInstanceName = "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})]
+        (_, LuaAlgBuilder{algLatestLuaValueInstance}) = runState (processStatement "_" assignment) defaultAlgBuilder
+     in algLatestLuaValueInstance @?= expected
+
+--case_process_multiple_assignments_statement =
+--    let assignment = Assign [VarName (Name $  "a"), VarName (Name $  "b")] [Number IntNum $  "2", Number FloatNum $  "2.5"]
+--        expected =
+--            Map.fromList
+--                [ ( "a", LuaValueInstance{luaValueInstanceName =  "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})
+--                , ( "b", LuaValueInstance{luaValueInstanceName =  "b", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False})
+--                ]
+--        (_str :: String, AlgBuilder{algLatestLuaValueInstance}) = runState (processStatement ( "_") assignment) defaultAlgBuilder
+--     in algLatestLuaValueInstance @?= expected
+
+-- | a = 1 + 2
+case_process_add_statement =
+    let assignment = Lua.Assign [Lua.VarName (Lua.Name "a")] [Lua.Binop Lua.Add (Lua.Number Lua.IntNum "1") (Lua.Number Lua.IntNum "2")]
+        expected =
+            [ LuaStatement{fIn = ["!1#0", "!2#0"], fOut = [LuaValueInstance{luaValueInstanceName = "a", luaValueInstanceAssignCount = 0, luaValueInstanceIsConstant = False}], fValues = [], fName = "add", fInt = []}
```

Comment:
Usage of constructors without explicit function should improve readability. 
`LuaStatement{fIn = ["!1#0", "!2#0"], fOut = [LuaValueInstance "a" 0 False], fValues = [], fName = "add", fInt = []} `

or maybe with a local for test function like: ‘luaVar “a” 0’